src/Sylius/Component/Core/Model/Customer.php

```diff
@@ -140,7 +140,6 @@ public function addAddress(AddressInterface $address)
     public function removeAddress(AddressInterface $address)
     {
         $this->addresses->removeElement($address);
-        $address->setCustomer(null);
```

Comment:
Cause all (at least IMO) those can't exists without it's parents. Exising without them polutes database with non-matching childrens.